cwf/gateway/fabfile.py

```diff
@@ -143,227 +127,213 @@ def integ_test(
+        c_cwf.local("mkdir -p " + tar_path)
+        output = c_cwf.local("docker images cwf_*").stdout
         for line in output.splitlines():
             if not line.startswith('cwf'):
                 continue
             line = line.rstrip("\n")
             image = line.split(" ")[0]
-            local("docker save -o %s/%s.tar %s" % (tar_path, image, image))
+            c_cwf.local(f"docker save -o {tar_path}/{image}.tar {image}")
 
-    output = local("ls %s/cwf_*.tar" % tar_path, capture=True)
+    output = c_cwf.local(f"ls {tar_path}/cwf_*.tar").stdout
     for line in output.splitlines():
-        regex = '%s/(.+?).tar' % tar_path
+        regex = f'{tar_path}/(.+?).tar'
         match = re.search(regex, line)
         if match:
             image = match.group(1)
-            put("%s/%s.tar" % (tar_path, image), "%s.tar" % image)
-            run('docker load -i %s.tar' % image)
+            c_cwf.put(f"{tar_path}/{image}.tar", f"{image}.tar")
+            c_cwf.run(f'docker load -i {image}.tar')
         else:
             print("We should not be here, but " + line + " does not match regex: " + regex)
 
 
-def _set_cwag_configs(configfile):
+def _set_cwag_configs(c_vm, configfile):
     """ Set the necessary config overrides """
-    with cd(CWAG_INTEG_ROOT):
-        sudo('mkdir -p /var/opt/magma')
-        sudo('mkdir -p /var/opt/magma/configs')
-        sudo("cp {} /var/opt/magma/configs/gateway.mconfig".format(configfile))
+    c_vm.sudo('mkdir -p /var/opt/magma/configs')
+    c_vm.sudo(f"cp {CWAG_INTEG_ROOT}/{configfile} /var/opt/magma/configs/gateway.mconfig")
 
 
-def _set_cwag_networking(mac):
-    sudo('arp -s %s %s' % (CWAG_TEST_IP, mac))
+def _set_cwag_networking(c_cwf, mac):
+    c_cwf.sudo(f'arp -s {CWAG_TEST_IP} {mac}')
 
 
-def _get_br_mac(bridge_name):
-    mac = run("cat /sys/class/net/%s/address" % bridge_name)
+def _get_br_mac(c_vm, bridge_name):
+    mac = c_vm.run(f"cat /sys/class/net/{bridge_name}/address").stdout.strip()
     return mac
 
 
-def _set_cwag_test_configs():
+def _set_cwag_test_configs(c_test):
     """ Set the necessary test configs """
-    sudo('mkdir -p /etc/magma')
+    c_test.sudo('mkdir -p /etc/magma')
     # Create empty uesim config
-    sudo('touch /etc/magma/uesim.yml')
+    c_test.sudo('touch /etc/magma/uesim.yml')
+    c_test.sudo('cp -r $MAGMA_ROOT/cwf/gateway/configs /etc/magma/')
 
 
-def _start_ipfix_controller():
+def _start_ipfix_controller(c_test):
     """ Start the IPFIX collector"""
-    with cd("$MAGMA_ROOT"):
-        sudo('mkdir -p records')
-        sudo('rm -rf records/*')
-        sudo('pkill ipfix_collector > /dev/null &', pty=False, warn_only=True)
-        sudo('tmux new -d \'/usr/bin/ipfix_collector -4tu -p 4740 -o /home/vagrant/magma/records\'')
+    with c_test.cd("$MAGMA_ROOT"):
+        c_test.run('mkdir -p records')
+        c_test.run('rm -rf records/*')
+        c_test.run('sudo pkill ipfix_collector > /dev/null &', pty=False, warn=True)
```

Comment:
NIT in line 327 `c_test.sudo` is used - why not here?

Reply:
The issue with `c.sudo` is that is does not work within a `c.cd` context (it fails with error `sudo: cd: command not found`). Writing `sudo` at the start of the `run` command is just a workaround (one could think of not using `c.sudo` at all for consistency)